pkg/agent/agent.go

```diff
@@ -405,11 +405,14 @@ func (i *Initializer) configureGatewayInterface(gatewayIface *interfacestore.Int
 
 func (i *Initializer) setupDefaultTunnelInterface(tunnelPortName string) error {
 	tunnelIface, portExists := i.ifaceStore.GetInterface(tunnelPortName)
+	localIP := i.getTunnelPortLocalIP()
 
 	// Check the default tunnel port.
 	if portExists {
 		if i.networkConfig.TrafficEncapMode.SupportsEncap() &&
-			tunnelIface.TunnelInterfaceConfig.Type == i.networkConfig.TunnelType {
+			tunnelIface.TunnelInterfaceConfig.Type == i.networkConfig.TunnelType &&
+			(tunnelIface.TunnelInterfaceConfig.LocalIP == nil && localIP == nil ||
```

Comment:
You can just do tunnelIface.LocalIP == localIP?

Reply:
`type IP []byte` The type of net.IP is slice, `slice can only be compared to nil`. So I can only compare the localIP with `nil`, and then use  `IP.Equal()`

Reply:
My mistake. But at least you can just do tunnelIface.TunnelInterfaceConfig.LocalIP.Equal(localIP) - it does not matter even tunnelIface.TunnelInterfaceConfig.LocalIP or localIP is nil.

Reply:
OK, Cool~